tests/process.h

```diff
@@ -0,0 +1,118 @@
+#include "unistd.h"
+#include "wait.h"
+#include <cassert>
+#include <experimental/filesystem>
+#include <optional>
+#include <signal.h>
+
+namespace fs = std::experimental::filesystem;
+
+namespace h5s3::testing {
+
+class process {
+
+public:
+    using environment = std::vector<std::pair<std::string, std::string>>;
+
+    enum class state {
+        RUNNING,
+        EXITED,
+        SIGNALED,
+    };
+
+    class status {
+    public:
+        int code;
+        state process_state;
+    };
+
+    class error : public std::runtime_error {
+    public:
+        explicit error(const std::string& message) : std::runtime_error(message) {}
+    };
+
+private:
+    std::vector<std::string> m_argv;
+    std::vector<std::string> m_envp;
+    pid_t m_pid;
+
+public:
+    process(std::vector<std::string>& args,
+            environment& env) {
+
+        // Initialize argv.
+        for (unsigned int i = 0; i < args.size(); ++i) {
```

Comment:
should we just use the vector copy constructor in the initializer list for this ctor?